### PR TITLE
Guncode Agony 4.4: Armor as an argument

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request
Lil cleanup/tweak I couldn't do in the main PR because it conflicted before and i forgot after. Yes this works with overrides that don't have the arg, yes I tested it.

## Why It's Good For The Game
Don't run armor code thrice please thank you

## Changelog
:cl:
code: Projectile impacts no longer fetch mobs' armor values thrice
/:cl:

### CODE REVIEW

File: code/game/atom/atom_act.dm [hunk 1]
```diff
@@ -83,26 +83,26 @@
  * * def_zone - zone hit
  * * piercing_hit - is this hit piercing or normal?
  */
-/atom/proc/bullet_act(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+/atom/proc/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE, blocked = check_projectile_armor(def_zone, proj))
```
Comment (LemonInTheDark) [inline code/game/atom/atom_act.dm:86]: please never ever do this just be explicit in the proc you want me dead

File: code/game/atom/atom_act.dm [hunk 2]
```diff
@@ -83,26 +83,26 @@
...
 	if(sigreturn & COMPONENT_BULLET_PIERCED)
 		return BULLET_ACT_FORCE_PIERCE
 	if(sigreturn & COMPONENT_BULLET_BLOCKED)
 		return BULLET_ACT_BLOCK
 	if(sigreturn & COMPONENT_BULLET_ACTED)
 		return BULLET_ACT_HIT
 
-	SEND_SIGNAL(src, COMSIG_ATOM_BULLET_ACT, hitting_projectile, def_zone)
-	if(QDELETED(hitting_projectile)) // Signal deleted it?
+	SEND_SIGNAL(src, COMSIG_ATOM_BULLET_ACT, proj, def_zone, piercing_hit, blocked)
```
Comment (LemonInTheDark) [inline code/game/atom/atom_act.dm:97]: update the docced args

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -91,12 +91,11 @@
 /mob/living/proc/is_ears_covered()
 	return null
 
-/mob/living/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE)
+/mob/living/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE, blocked = check_projectile_armor(def_zone, proj))
```
Comment (LemonInTheDark) [inline code/modules/mob/living/living_defense.dm:94]: similar (that or make a wrapper proc for bullet_act that sets this var and also does all the signal stuff)

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/atom/atom_act.dm (hunk 1 above)
Comment (MrMelbert) [reply code/game/atom/atom_act.dm]: I think it's based

File: code/game/atom/atom_act.dm [hunk 4]
```diff
@@ -75,34 +75,47 @@
...
+ * Wrapper for bullet_act used for atom-specific calculations, i.e. armor
+ *
+ * @params
+ * * hitting_projectile - projectile
+ * * def_zone - zone hit
+ * * piercing_hit - is this hit piercing or normal?
+ */
+
+/atom/proc/projectile_hit(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+	return bullet_act(hitting_projectile, def_zone, piercing_hit, check_projectile_armor(def_zone, hitting_projectile))
```
Comment (LemonInTheDark) [inline code/game/atom/atom_act.dm:88]: shouldn't this still take blocked damage as input to allow for an override of the proc?

**Pull request merged**